src/python/ddapp/sceneloader.py

```diff
@@ -54,5 +54,81 @@ def loadSDF(self, filename):
                     self.affordanceManager.newAffordanceFromDescription(desc)
                 if col.geometry_type=='cylinder':
                     desc = dict(classname='CylinderAffordanceItem', Name=name, uuid=newUUID(), pose=p, Color=[0.8, 0.8, 0.8], Radius=float(col.geometry_data['radius']), Length = float(col.geometry_data['length']))
-                    self.affordanceManager.newAffordanceFromDescription(desc)            
+                    self.affordanceManager.newAffordanceFromDescription(desc)
+                    
+    def generateSDFfromAffordances(self):
+        filename= os.environ['DRC_BASE'] + '/software/models/worlds/directorAffordances.sdf'
+        sdfFile = open(filename, 'w')
+        am = segmentation.affordanceManager
+        affordances = am.getAffordances()
+        xmlStr = '<sdf version=\'1.4\'>\n'\
+                 '\t<world name=\'directorAffordances\'>\n'
+        for aff in affordances:
+            if aff.getDescription()['classname'] in ['BoxAffordanceItem', 'CylinderAffordanceItem']:
+                name = aff.getDescription()['Name']
+                xmlStr += '\t\t<model name=\'{0:s}\'>\n'\
+                          '\t\t\t<pose>0 0 0 0 0 0</pose>\n'\
+                          '\t\t\t<link name=\'{0:s}\'>\n'.format(name)
+                pose = np.append(aff.getDescription()['pose'][0], transformUtils.quaternionToRollPitchYaw(aff.getDescription()['pose'][1]))
+                xmlStr += '\t\t\t\t<pose>{:s}</pose>\n'.format(' '.join(map(str, pose)))
+                collisions = aff.getDescription()['Collision Enabled']
+                color = aff.getDescription()['Color']
+                alpha = aff.getDescription()['Alpha']
+                if aff.getDescription()['classname'] == 'BoxAffordanceItem':
+                    geometry = 'box'
+                    dimensions = aff.getDescription()['Dimensions']
+                    if collisions:
+                        xmlStr += '\t\t\t\t<collision name=\'collision\'>\n'\
```

Comment:
you might consider using an xml library to generate the output instead of string formatting.
also, why not spaces instead of all the tabs?